github/RepositoryAdvisory.py

```diff
@@ -326,97 +279,75 @@ def clear_credits(self):
 
     def edit(
         self,
-        summary: github.GithubObject.Opt[str] = github.GithubObject.NotSet,
-        description: github.GithubObject.Opt[str] = github.GithubObject.NotSet,
-        severity_or_cvss_vector_string: github.GithubObject.Opt[
-            str
-        ] = github.GithubObject.NotSet,
-        cve_id: github.GithubObject.Opt[str] = github.GithubObject.NotSet,
-        vulnerabilities: github.GithubObject.Opt[
-            typing.Iterable[AdvisoryVulnerability]
-        ] = github.GithubObject.NotSet,
-        cwe_ids: github.GithubObject.Opt[
-            typing.Iterable[str]
-        ] = github.GithubObject.NotSet,
-        credits: github.GithubObject.Opt[
-            typing.Iterable[Credit]
-        ] = github.GithubObject.NotSet,
-        state: github.GithubObject.Opt[str] = github.GithubObject.NotSet,
-    ) -> "RepositoryAdvisory":
+        summary: Opt[str] = NotSet,
+        description: Opt[str] = NotSet,
+        severity_or_cvss_vector_string: Opt[str] = NotSet,
+        cve_id: Opt[str] = NotSet,
+        vulnerabilities: Opt[Iterable[AdvisoryVulnerability]] = NotSet,
+        cwe_ids: Opt[Iterable[str]] = NotSet,
+        credits: Opt[Iterable[Credit]] = NotSet,
+        state: Opt[str] = NotSet,
+    ) -> RepositoryAdvisory:
         """
         :calls: `PATCH /repos/{owner}/{repo}/security-advisories/:advisory_id <https://docs.github.com/en/rest/security-advisories/repository-advisories>`_
-        :param summary: string
-        :param description: string
-        :param severity_or_cvss_vector_string: string
-        :param cve_id: string
-        :param vulnerabilities: iterable of :class:`github.RepositoryAdvisoryVulnerability.AdvisoryVulnerability`
-        :param cwe_ids: iterable of string
-        :param credits: iterable of :class:`github.RepositoryAdvisoryCredit.Credit`
-        :param state: string
-        :rtype: :class:`github.RepositoryAdvisory.RepositoryAdvisory`
         """
-        assert summary is github.GithubObject.NotSet or isinstance(
-            summary, str
-        ), summary
-        assert description is github.GithubObject.NotSet or isinstance(
-            description, str
-        ), description
-        assert (
-            severity_or_cvss_vector_string is github.GithubObject.NotSet
-            or isinstance(severity_or_cvss_vector_string, str)
-        ), (severity_or_cvss_vector_string)
-        assert cve_id is github.GithubObject.NotSet or isinstance(cve_id, str), cve_id
-        assert vulnerabilities is github.GithubObject.NotSet or isinstance(
-            vulnerabilities, typing.Iterable
+        assert summary is NotSet or isinstance(summary, str), summary
```

Comment:
this can be refactored with new type-assert utils, but I'm hoping not to touch much logic in this PR